ユーザー管理機能

kicucokko opened the pull request:
**# What**
-
ユーザー管理機能の実装

**# Why**
-
新規登録・ログイン・エラー表示についての挙動確認

[新規登録](https://i.gyazo.com/93b6e0329ad527bddedaa272ec8de46c.gif)
[新規登録時エラー画面](https://i.gyazo.com/c53bc549739298c5d11a0a38030f225f.gif)
[ログイン画面](https://i.gyazo.com/ab865027da14e236d636a833c4688dee.gif)
[ログアウト時画面](https://i.gyazo.com/136181558fce60e7d5496ed2ececcde8.gif)
[ログイン時エラー画面](https://i.gyazo.com/18650c2b475d20130063d7b6bb09a333.gif)
[ログアウト時画面](https://i.gyazo.com/136181558fce60e7d5496ed2ececcde8.gif)
![テスト成功画像](https://i.gyazo.com/747298ebf31a3b0de70ce5ad0a1fc500.jpg)

kicucokko committed:
deviseの導入

kicucokko committed:
カラム作成とユーザー登録画面構築

・usersテーブルの作成とバリテーションを記述
・ユーザー登録画面表示のためのcontroller/viewsへの記述
・新規登録画面表示のための部分テンプレートの挿入

kicucokko committed:
トップページ導入とユーザー登録機能の実装

・items/index.html.erbの導入
・トップ画面〜新規登録のパス設定
・usersテーブルへの登録実装
・ログイン前と登録完了後のトップページヘッダーの切替

kicucokko committed:
ログイン機能の実装

kicucokko committed:
エラーメッセージの設定・バリデーションの記述

・エラーメッセージテンプレートの挿入
・新規登録画面について
　パスワードが6文字以上
　名前は全角文字であること
　カナは全角カタカナであること
以上のバリデーション記述
・バリデーションに伴うエラーメッセージの修正

kicucokko committed:
単体テスト準備

・rspec/Factrybot/Fakerの導入
・必要ファイルの作成

kicucokko committed:
ユーザー新規登録の単体テスト実施

kicucokko committed:
レイアウト微修正

maimai0401 commented on spec/models/user_spec.rb:
```diff
@@ -0,0 +1,106 @@
+        @user.valid?
+        expect(@user.errors.full_messages).to include("Password is too short (minimum is 6 characters)")
+      end
+      it 'passwordに半角英数字が混合していないと登録できない' do
```
より正確なテストを行うために、下記テストコードに分けて追加で実装しましょう。
・passwordは英語のみでは登録できないこと
・passwordは数字のみでは登録できないこと
・passwordは全角では登録できないこと

maimai0401 reviewed (commented):
レビュー依頼ありがとうございます。
1点コメントいたしましたので、ご確認をお願いいたします。
修正後、再度レビュー依頼フォームにてご提出くださいませ。

kicucokko committed:
テストコード追加記述

・password：数字、アルファベット、全角のいづれか単一のみでは登録できない

kicucokko commented:
**# What**
-
ユーザー管理機能の実装

**# Why**
-
新規登録・ログイン・エラー表示についての挙動確認

**前回からの修正点**
Passwordに関するテストコードの追加×3項目：数字、アルファベット、全角いづれか単一では登録できない

ご確認よろしくお願いいたします。
![テスト実行画像(改)](https://i.gyazo.com/a685e4197e6633c7489048d321945aef.png)

ushi10 commented on app/models/user.rb:
```diff
@@ -0,0 +1,20 @@
+    validates :last_name
+  end
+
+  with_options presence: true, format: { with: /\A[ァ-ヶ]+\z/ ,message: "Input full-width katakana characters"} do
```
```ruby
/\A[ァ-ヶ]+\z/
```
記載の正規表現ですと「メアリー」のような「ー」を認識してくれません。
上記内容も保存できるよう正規表現を書き換えてみましょう。

ushi10 reviewed (commented):
レビュー依頼ありがとうございます！ 修正点確認させていただきました。
細かい点ではありますが追加でコメントいたしましたので、ご確認のほどよろしくお願いいたします。
修正頂いた後、コミットプッシュを確認いただき、再度レビュー依頼のご提出をお願い致します！

kicucokko committed:
正規表現の微修正

・last_kana,first_kana：Userモデル正規表現の修正：”ー”の追加
・テストコードFactoryBotのテストデータ一部変更：first_kana→タロー

kicucokko commented:
# What
ユーザー管理機能の実装

# Why
新規登録・ログイン・エラー表示についての挙動確認

前回からの修正点
・Userモデルの修正：last_kana,first_kanaの正規表現へ”ー”を追記
・上記に伴いテストコードFactoryBotデータ(first_kana)の変更：タロウ→タロー

ご確認よろしくお願いいたします。
![テスト実施画像(改2)](https://i.gyazo.com/16fff722de195ca5a4827c4e40422810.png)

ushi10 reviewed (commented):
レビュー依頼ありがとうございます！確認させていただきました。 
こちら、LGTMとさせていただきます。実装お疲れさまでした！

改めて挙動を確認していただき、問題がない様子でしたらこちらのブランチを忘れずマージしましょう。
マージ後、ローカルのmasterブランチにて忘れずにプルをしましょう。 その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。

引き続き開発を頑張っていきましょう！
